src/draw_partial.jl

```diff
@@ -0,0 +1,474 @@
+                push!(co_states, true)
+                if last(pointslist) == first(pointslist)
+                    # don’t repeat first point, we can close it ourselves
+                    if length(pointslist) > 2
+                        pop!(pointslist)
+                    end
+                end
+                push!(polygonlist, pointslist)
+                pointslist = Point[]
+            else
+                error("pathtopoly(): unknown CairoPathEntry " * repr(e.element_type))
+                error("pathtopoly(): unknown CairoPathEntry " * repr(e.points))
+            end
+        end
+        # the path was never closed, so flush
+        if length(pointslist) > 0
+            push!(co_states, false)
+            push!(polygonlist, pointslist)
+        end
+    end
+    #"""check if everything went well"""
+    @assert length(polygonlist) == length(co_states)
+    #"""return polygonlist, and its closed/open state"""
+    return polygonlist, co_states
+end
+
+##CONTEXT strokelength 
+#here are a bunch of overdubs to calculate strokelength
+#strokepath
+function overdub(c::ctx_strokelength, ::typeof(Luxor.strokepath), args...)
+    polys, co = pathtopoly(:yes)
+    len = sum([polyperimeter(p, closed = c) for (p, c) in zip(polys, co)])
+    push!(dp_state.cur_len_perim, len)
+    push!(
+        dp_state.stroke_polydists,
+        [polydistances(p, closed = c) for (p, c) in zip(polys, co)],
+    )
+    newpath() #strokepath makes a newpath
+    nothing
+end
+
+#strokepreserve
+function overdub(c::ctx_strokelength, ::typeof(Luxor.strokepreserve), args...)
+    polys, co = pathtopoly(:yes)
+    len = sum([polyperimeter(p, closed = co) for (p, co) in zip(polys, co)])
+    push!(dp_state.cur_len_perim, len)
+    push!(
+        dp_state.stroke_polydists,
+        [polydistances(p, closed = c) for (p, c) in zip(polys, co)],
+    )
+    nothing
+end
+
+#fillpath
+#=
+fillpath will be animated by "painting" in the path 
+a growing circle starting at one corner of the path's
+bounding box to the other end, therefore the diagonal
+of the BoudingBox is added to the perimeter 
+=#
+function overdub(c::ctx_strokelength, ::typeof(Luxor.fillpath), args...)
+    poly1, _ = pathtopoly(:yes)
+    if length(vcat(poly1...)) > 2
+        bbox = BoundingBox(vcat(poly1...))
+        push!(dp_state.cur_len_perim, distance(bbox[1], bbox[2]))
+    else
+        push!(dp_state.cur_len_perim, 0)
+    end
+    newpath() #fillpath makes a new path
+end
+
+#fillpreserve
+function overdub(c::ctx_strokelength, ::typeof(Luxor.fillpreserve), args...)
+    poly1, _ = pathtopoly(:yes)
+    if length(vcat(poly1...)) > 2
+        bbox = BoundingBox(vcat(poly1...))
+        push!(dp_state.cur_len_perim, distance(bbox[1], bbox[2]))
+    else
+        push!(dp_state.cur_len_perim, 0)
+    end
+end
+
+#hue
+#="""for some reason parsing colors fails to furthur overdub , so we just
+return as is"""=#
+function overdub(c::ctx_strokelength, ::typeof(Colors.parse), args...)
```

Comment:
Colors.parse simply doesn't play well with Cassette , might be something related to this https://github.com/JuliaLabs/Cassette.jl/issues/130